crates/shared/src/sources/balancer_v2/pool_cache.rs

```diff
@@ -31,9 +34,14 @@ impl PoolReserveFetcher {
         common_pool_fetcher: Arc<dyn common::PoolInfoFetching>,
         web3: Web3,
     ) -> Result<Self> {
+        let weighted_pool_factory = BalancerV2WeightedPoolFactory::deployed(&web3).await?;
```

Comment:
Is it the plan to keep this component manage one instance of each pool type factory or will this eventually also be united behind a single component per pool type?

Reply:
Correct. We will make an object-safe interface per factory (i.e. `Arc<dyn Thing>` - what I will call a `Registry`) that we can aggregate in a list `Vec<Arc<dyn Thing>>` (what I will call `AggregatePoolFetcher`). This way, adding new pools envolves:
- defining info and pool fetching details (i.e. implement `FactoryIndexing`)
- instantiating a new `Registry` component and adding it to our `AggregateFetcher`'s its list of registries.